### PR TITLE
{2023.06}[foss/2023b] GROMACS 2024.1

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
Build & test of GROMACS worked on `haswell`, but there's a bug introduced by #494 causing trouble:

```
ERROR: One or more files not found: GROMACS-2024.1-foss-2023b.eb
```

`make test` of GROMACS failed on `neoverse_v1`:
```
The following tests FAILED:
         15 - GmxlibTests (Subprocess aborted)
         46 - SimdUnitTests (Failed)
         81 - MdrunFEPTests (Failed)
Errors while running CTest
```

<details>

```
15/87 Test #15: GmxlibTests ..................................Subprocess aborted***Exception:   0.58 sec
[==========] Running 78 tests from 2 test suites.
[----------] Global test environment set-up.
[----------] 72 tests from NBInteraction/NonbondedFepTest
[ RUN      ] NBInteraction/NonbondedFepTest.testKernel/0
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/testutils/refdata.cpp:957: Failure
   In item: /Forces/[1]/X
    Actual: -450.75637494131161
 Reference: -94.898901947422004
Difference: 355.857 (10259072709535721 double-prec. ULPs, rel. 3.75)
 Tolerance: abs. 1e-08, rel. 1e-11, 100 ULPs
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/testutils/refdata.cpp:957: Failure
   In item: /Forces/[2]/X
    Actual: 189.79780389484378
 Reference: 94.898901947421891
Difference: 94.8989 (4503599627370496 double-prec. ULPs, rel. 1)
 Tolerance: abs. 1e-08, rel. 1e-11, 100 ULPs
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/testutils/refdata.cpp:957: Failure
   In item: /Forces/[3]/X
    Actual: 521.91714209293536
 Reference: 260.95857104646768
Difference: 260.959 (4503599627370496 double-prec. ULPs, rel. 1)
 Tolerance: abs. 1e-08, rel. 1e-11, 100 ULPs
[  FAILED  ] NBInteraction/NonbondedFepTest.testKernel/0, where GetParam() = (4-byte object <00-00 00-00>, 936-byte object <25-00 00-00 BD-37 86-35 3A-8C 30-E2 8E-79 45-3E 03-00 00-00 15-BD 63-BF 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 10-98 C5-2B 00-00 00-00 D0-9E C5-2B 00-00 00-00 ... 33-33 33-33 33-33 EB-3F 33-33 33-33 33-33 D3-3F 33-33 33-33 33-33 D3-3F F3-21 A8-1A BD-DA 02-00 00-00 00-00 00-00 00-00 00-00 00-00 F1-30 62-BF 00-00 00-00 00-00 00-00 00-00 00-00 00-00 F0-3F>, { 0x2bc34900, 0x2bc34918, 0x2bc34930, 0x2bc34948 }, 0, 0, true) (0 ms)
[ RUN      ] NBInteraction/NonbondedFepTest.testKernel/1
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/testutils/refdata.cpp:957: Failure
   In item: /Forces/[1]/X
    Actual: -450.75637494131161
 Reference: -94.898901947422004
Difference: 355.857 (10259072709535721 double-prec. ULPs, rel. 3.75)
 Tolerance: abs. 1e-08, rel. 1e-11, 100 ULPs
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/testutils/refdata.cpp:957: Failure
   In item: /Forces/[2]/X
    Actual: 189.79780389484378
 Reference: 94.898901947421891
Difference: 94.8989 (4503599627370496 double-prec. ULPs, rel. 1)
 Tolerance: abs. 1e-08, rel. 1e-11, 100 ULPs
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/testutils/refdata.cpp:957: Failure
   In item: /Forces/[3]/X
    Actual: 521.91714209293536
 Reference: 260.95857104646768
Difference: 260.959 (4503599627370496 double-prec. ULPs, rel. 1)
 Tolerance: abs. 1e-08, rel. 1e-11, 100 ULPs
free(): invalid next size (fast)
```

```
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/vdwalone_d, where GetParam() = ("vdwalone", 1, { 87 }) (85 ms)
[----------] 12 tests from EquivalentToReference/FreeEnergyReferenceTest (1178 ms total)

[----------] Global test environment tear-down
[==========] 12 tests from 1 test suite ran. (1203 ms total)
[  PASSED  ] 2 tests.
[  FAILED  ] 10 tests, listed below:
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwsequential_coul_d, where GetParam() = ("coulandvdwsequential_coul", 1, { 89, 90 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwsequential_vdw_d, where GetParam() = ("coulandvdwsequential_vdw", 1, { 89, 90 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_d, where GetParam() = ("coulandvdwtogether", 1, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_net_charge_d, where GetParam() = ("coulandvdwtogether-net-charge", 2, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_decouple_counter_charge_d, where GetParam() = ("coulandvdwtogether-decouple-counter-charge", 1, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/expanded_d, where GetParam() = ("expanded", 1, { 89, 90 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/relative_d, where GetParam() = ("relative", 11, { 87, 89, 90, 91 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/relative_position_restraints_d, where GetParam() = ("relative-position-restraints", 11, { 87, 89, 90, 91, 92 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/transformAtoB_d, where GetParam() = ("transformAtoB", 1, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/vdwalone_d, where GetParam() = ("vdwalone", 1, { 87 })

10 FAILED TESTS
```

```
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3Overlapping
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterDecrU3
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterDecrU3Overlapping

 4 FAILED TESTS
```

```
[ RUN      ] SimdFloatingpointUtilTest.transposeScatterIncrU3Overlapping
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/src/gromacs/simd/tests/simd_floatingpoint_util.cpp:393: Failure
  Value of: mem0_[j]
    Actual: 2015.0000000000446
  Expected: refmem[j]
  Which is: 1003.0000000000223
Difference: 1012 (4543182045970432 double-prec. ULPs, rel. 1.01)
 Tolerance: abs. 8.88178e-16, 4 ULPs
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3Overlapping (0 ms)
```

</details>

**Comment by boegel** [comment]
For `zen3` (job `8077`) and `skylake_avx512` (job `8075`), we again saw a failing test:

```
The following tests FAILED:
          2 - GmxapiMpiTests (Failed)
```

due to:

```
File input/output error:
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/easybuild_obj/api/gmxapi/cpp/tests/Testing/Temporary/GmxApiTest_RunnerChainedMD.trr
```

**Comment by ocaisa** [comment]
> Build & test of GROMACS worked on `haswell`, but there's a bug introduced by #494 causing trouble:

I think it's actually OpenMPI giving flaky behaviour due to the bug we found, and it would/could be fixed by #500 .

I do (also) believe there is a bug in #494 though, as it _should_ be reporting that the EB PR for GROMACS is not merged (but that may also be due to  #501)

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
I just want to check if a rebuild gives a fluke success...perhaps there is a race condition and we need to reduce parallelism

**Comment by casparvl** [comment]
> I do (also) believe there is a bug in https://github.com/EESSI/software-layer/pull/494 though, as it should be reporting that the EB PR for GROMACS is not merged (but that may also be due to https://github.com/EESSI/software-layer/pull/501)

Almost surely due to #501. #494 only reports this reason if the `eb -missing` _passes_ when we don't strip the `--from-pr`'s. Due to #501 , `eb --missing` probably still fails, even with `--from-pr`s still in. Essentially, that means the "Reason" from #494  is only printed if the unmerged PR is the _only_ issue left.

**Comment by ocaisa** [comment]
@boegel So there are random failures at play here, the Zen3 build [succeeded on the second try](https://github.com/EESSI/software-layer/pull/499#issuecomment-1996836413). I still believe this is due to the OpenMPI bug and could be fixed by either #500 or #488 (preferred)

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
Unfortunately `aarch64-neoverse_v1` is a persistent failure

**Comment by boegel** [comment]
@ocaisa Maybe a good next step is to check whether we're seeing the same problem outside of the EESSI build environment on `neoverse_v1`?

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
@casparvl It still seems to not be catching the reason as expected: https://github.com/EESSI/software-layer/pull/499#issuecomment-1999827233

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
Unfortunately there are real issues with `aarch64-neoverse_v1`, I've [opened an issue with GROMACS](https://gitlab.com/gromacs/gromacs/-/issues/5057) and hopefully they help us pin it down

**Comment by boegel** [comment]
Updated PR to use easyconfig from https://github.com/easybuilders/easybuild-easyconfigs/pull/20439, which includes patch to apply fix from https://gitlab.com/gromacs/gromacs/-/merge_requests/4066

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
Still test failures on Neoverse V1, similar to what was reported by @ocaisa in https://gitlab.com/gromacs/gromacs/-/issues/5057

<details>

```
[bot@login1 Temporary]$ grep FAILED LastTest.log
[  FAILED  ] NBInteraction/NonbondedFepTest.testKernel/0, where GetParam() = (4-byte object <00-00 00-00>, 936-byte object <25-00 00-00 BD-37 86-35 3A-8C 30-E2 8E-79 45-3E 03-00 00-00 15-BD 63-BF 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 00-00 10-98 66-3C 00-00 00-00 D0-9E 66-3C 00-00 00-00 ... 33-33 33-33 33-33 EB-3F 33-33 33-33 33-33 D3-3F 33-33 33-33 33-33 D3-3F F3-21 A8-1A BD-DA 02-00 00-00 00-00 00-00 00-00 00-00 00-00 F1-30 62-BF 00-00 00-00 00-00 00-00 00-00 00-00 00-00 F0-3F>, { 0x3c644900, 0x3c644918, 0x3c644930, 0x3c644948 }, 0, 0, true) (0 ms)
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3 (0 ms)
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3Overlapping (0 ms)
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterDecrU3 (0 ms)
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterDecrU3Overlapping (0 ms)
[  FAILED  ] 4 tests, listed below:
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterIncrU3Overlapping
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterDecrU3
[  FAILED  ] SimdFloatingpointUtilTest.transposeScatterDecrU3Overlapping
 4 FAILED TESTS
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwsequential_coul_d, where GetParam() = ("coulandvdwsequential_coul", 1, { 89, 90 }) (110 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwsequential_vdw_d, where GetParam() = ("coulandvdwsequential_vdw", 1, { 89, 90 }) (98 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_d, where GetParam() = ("coulandvdwtogether", 1, { 87 }) (95 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_net_charge_d, where GetParam() = ("coulandvdwtogether-net-charge", 2, { 87 }) (55 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_decouple_counter_charge_d, where GetParam() = ("coulandvdwtogether-decouple-counter-charge", 1, { 87 }) (42 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/expanded_d, where GetParam() = ("expanded", 1, { 89, 90 }) (184 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/relative_d, where GetParam() = ("relative", 11, { 87, 89, 90, 91 }) (115 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/relative_position_restraints_d, where GetParam() = ("relative-position-restraints", 11, { 87, 89, 90, 91, 92 }) (115 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/transformAtoB_d, where GetParam() = ("transformAtoB", 1, { 87 }) (102 ms)
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/vdwalone_d, where GetParam() = ("vdwalone", 1, { 87 }) (88 ms)
[  FAILED  ] 10 tests, listed below:
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwsequential_coul_d, where GetParam() = ("coulandvdwsequential_coul", 1, { 89, 90 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwsequential_vdw_d, where GetParam() = ("coulandvdwsequential_vdw", 1, { 89, 90 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_d, where GetParam() = ("coulandvdwtogether", 1, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_net_charge_d, where GetParam() = ("coulandvdwtogether-net-charge", 2, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/coulandvdwtogether_decouple_counter_charge_d, where GetParam() = ("coulandvdwtogether-decouple-counter-charge", 1, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/expanded_d, where GetParam() = ("expanded", 1, { 89, 90 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/relative_d, where GetParam() = ("relative", 11, { 87, 89, 90, 91 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/relative_position_restraints_d, where GetParam() = ("relative-position-restraints", 11, { 87, 89, 90, 91, 92 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/transformAtoB_d, where GetParam() = ("transformAtoB", 1, { 87 })
[  FAILED  ] EquivalentToReference/FreeEnergyReferenceTest.WithinTolerances/vdwalone_d, where GetParam() = ("vdwalone", 1, { 87 })
10 FAILED TESTS
```

</details>

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1

**Comment by boegel** [comment]
Hmm, interesting (new?) test failure for `x86_64/generic`:

```
99% tests passed, 1 tests failed out of 91

...

The following tests FAILED:
          2 - GmxapiMpiTests (Failed)
```

<details>

```
Test failures from rank 0:
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/api/gmxapi/cpp/tests/stopsignaler.cpp:206: Failure
Expected equality of these values:
  nsteps
    Which is: 4
  gmx::roundToInt(restraint->timeElapsedSinceStart() / getTestStepSize())
    Which is: 0
Test failures from rank 1:
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/gromacs-2024.1/api/gmxapi/cpp/tests/stopsignaler.cpp:206: Failure
Expected equality of these values:
  nsteps
    Which is: 4
  gmx::roundToInt(restraint->timeElapsedSinceStart() / getTestStepSize())
    Which is: 0
^[[0;31m[  FAILED  ] ^[[mGmxApiTest.ApiRunnerStopSignalClient (245 ms)
^[[0;32m[ RUN      ] ^[[mGmxApiTest.SystemConstruction
```

</details>

**Comment by bedroge** [comment]
> Hmm, interesting (new?) test failure for `x86_64/generic`:
> 
> ```
> 99% tests passed, 1 tests failed out of 91
> 
> ...
> 
> The following tests FAILED:
>           2 - GmxapiMpiTests (Failed)
> ```

@boegel I also saw that same failure on a haswell node with 2023.4:
https://github.com/EESSI/software-layer/pull/463#issuecomment-2037111093

**Comment by ocaisa** [comment]
Already a number of cases where the `x86-generic` build succeeded (for example https://github.com/EESSI/software-layer/pull/499#issuecomment-2038263386), so I triggered a rebuild to see if it succeeds

**Comment by boegel** [comment]
> > Hmm, interesting (new?) test failure for `x86_64/generic`:
> > ```
> > 99% tests passed, 1 tests failed out of 91
> > 
> > ...
> > 
> > The following tests FAILED:
> >           2 - GmxapiMpiTests (Failed)
> > ```
> 
> @boegel I also saw that same failure on a haswell node with 2023.4: [#463 (comment)](https://github.com/EESSI/software-layer/pull/463#issuecomment-2037111093)

Seems like another flaky failure that we should report upstream?
Let's first see if it's consistent, since meanwhile I'm working on providing more detailed feedback for the failing tests on Neoverse V1 anyway...

**Comment by boegel** [comment]
Last failure for `x86_64/generic` is also for `GmxapiMpiTests`, but seems different, actual failure is:
```
File input/output error:
/tmp/bot/easybuild/build/GROMACS/2024.1/foss-2023b/easybuild_obj/api/gmxapi/cpp/tests/Testing/Temporary/GmxApiTest_RunnerChainedMD.trr
```

This seems like the sort of problem that https://gitlab.com/gromacs/gromacs/-/merge_requests/4066 is supposed to fix...
Maybe the fix is incomplete?
I'll ask in https://gitlab.com/gromacs/gromacs/-/issues/4654

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic

**Comment by boegel** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
3rd time is a charm for `x86_64/generic`, and `aarch64/neoverse_v1` should pass too now that we're using `-DGMX_SIMD=ARM_NEON_ASIMD` there for GROMACS <= 2024.1

The last `neoverse_v1` build (job `10032`) failed seemingly due to interference of the manual ingestion of the rebuilds for #546

**Comment by bedroge** [comment]
bot: status

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1

**Comment by boegel** [comment]
@ocaisa @bedroge Ready to deploy! 🥳

File: eb_hooks.py [hunk 1]
```diff
@@ -335,6 +335,21 @@ def pre_configure_hook(self, *args, **kwargs):
...
+    """
+    Pre-configure hook for GROMACS:
+    - avoid building with SVE instructions on Neoverse V1 as workaround for failing tests,
+      see https://gitlab.com/gromacs/gromacs/-/issues/5057 + https://gitlab.com/eessi/support/-/issues/47
+    """
+    if self.name == 'GROMACS':
+        cpu_target = get_eessi_envvar('EESSI_SOFTWARE_SUBDIR')
+        if LooseVersion(self.version) <= LooseVersion('2024.1') and cpu_target == CPU_TARGET_NEOVERSE_V1:
+            self.cfg.update('configopts', '-DGMX_SIMD=ARM_NEON_ASIMD')
+            print_msg("Avoiding use of SVE instructions for GROMACS %s by using ARM_NEON_ASIMD as GMX_SIMD value", self.version)
```
Comment (bedroge) [inline eb_hooks.py:348]: @boegel Should we also add this to the known issues file?

Comment (boegel) [reply eb_hooks.py:348]: yes, done in 3eee364

**Review by bedroge — APPROVED** [review]

**Review by ocaisa — APPROVED** [review]

**Pull request merged**